docs(design): add latency definition, records service and visualization (#106)
tier4/CARET_doc: docs/design/latency_definitions/callback.md

# Callback

> The callback latency is the time difference from callback start to callback end.

Reviewer: ```suggestion
Callback latency is defined as duration between beginning and end of callback execution.
```

Reviewer: `Time difference` has meaning for expressing difference between two time zone. It is not appropriate word here.
https://www.collinsdictionary.com/jp/dictionary/english/time-difference

Author: addressed: [ea6f556](https://github.com/tier4/CARET_doc/pull/106/commits/ea6f556a84802265fdd6222bee05a63aa0620a68)

$$

> l_{\rm{callback}} = t_{\rm{callback\ end}} - t_{\rm{callback\ start}}

Reviewer 2: reloaded, then showed up correctly.

$$

> A simplified sequence diagram focusing only on the relevant data flow is shown below.

Reviewer: I think that it is confusing the sequence diagram is designed to show data flow. 

```suggestion
The sequence diagram shows how CARET rclcpp pick up two events; callback_start and callback_end 
```

Author: addressed: [ea6f556](https://github.com/tier4/CARET_doc/pull/106/commits/ea6f556a84802265fdd6222bee05a63aa0620a68)

```plantuml

participant "UserCode \n Callback" as Callback
participant "rclcpp" as Rclcpp
participant LTTng

activate Rclcpp

> Rclcpp -> LTTng : callback start

Reviewer: sample `callback_start`

> Rclcpp -> LTTng : callback start

Reviewer: ```suggestion
Rclcpp -> LTTng : sample callback_start
```

Author: addressed: [ea6f556](https://github.com/tier4/CARET_doc/pull/106/commits/ea6f556a84802265fdd6222bee05a63aa0620a68)

Rclcpp -> Callback : execute callback

activate Callback
Callback -> Rclcpp
deactivate Callback

> Rclcpp -> LTTng : callback end

Reviewer: sample callback_end

> Rclcpp -> LTTng : callback end

Reviewer: ```suggestion
Rclcpp -> LTTng : sample callback_end
```

Author: addressed: [ea6f556](https://github.com/tier4/CARET_doc/pull/106/commits/ea6f556a84802265fdd6222bee05a63aa0620a68)

# Callback

Callback latency is defined as duration between beginning and end of callback execution.
These events are represented as `callback_start` and `callback_end`, respectively.

$$
l_{\rm{callback}} = t_{\rm{callback\ end}} - t_{\rm{callback\ start}}
$$

> The sequence diagram shows how CARET rclcpp pick up two events; callback_start and callback_end.

Reviewer: I'm sorry, but I took a simple mistake.

```suggestion
The sequence diagram shows how CARET rclcpp picks up two events; callback_start and callback_end.
```

Author: addressed

```plantuml

participant "UserCode \n Callback" as Callback
participant "rclcpp" as Rclcpp
participant LTTng

activate Rclcpp

Rclcpp -> LTTng : sample callback_start
Rclcpp -> Callback : execute callback

activate Callback
Callback -> Rclcpp
deactivate Callback
Rclcpp -> LTTng : sample callback_end
```

> `to_dataframe` API returns following columns.

Reviewer: ```suggestion
`to_dataframe` API returns a table which has the following columns.
```

Author: addressed

Author: addressed
